core/service/src/components.rs

```diff
@@ -390,6 +404,8 @@ pub trait Components: Sized + 'static {
 	type ImportQueue: ImportQueue<FactoryBlock<Self::Factory>> + 'static;
 	/// The Fork Choice Strategy for the chain
 	type SelectChain: SelectChain<FactoryBlock<Self::Factory>>;
+	/// The authority key provider.
+	type AuthorityKeyProvider: offchain::AuthorityKeyProvider + Send + Sync;
```

Comment:
Can you elaborate why it needs to be part of `Service`?

Reply:
We need to know the AuraPair and GrandpaPair which aren't known yet. We could add those as types instead.

Reply:
Yeah, I think adding the key types to components/service would be better than allowing to customize the entire key provider.